docs/src/main/tut/docs/02-A-First-Example.md

```diff
@@ -0,0 +1,158 @@
+final case class SubtractF[A](a: A, b: A) extends ArithmeticF[A]
+final case class MultiplyF[A](a: A, b: A) extends ArithmeticF[A]
+final case class DivideF[A](a: A, b: A) extends ArithmeticF[A]
+```
+
+So, everywhere that referred to `Arithmetic` before now just refers to `A`, and we can fill that with whatever type we want. And, with a type parameter, come the type class instances (this should be derivable, but doesn’t yet exist in shapeless-contrib)
+
+```tut:book
+import scalaz.{Divide => _, _}, Scalaz._
+
+implicit val traverse: Traverse[ArithmeticF] = new Traverse[ArithmeticF] {
+  def traverseImpl[G[_], A, B]
+    (fa: ArithmeticF[A])
+    (f: A => G [B])
+    (implicit G: Applicative[G]) =
+    fa match {
+      case NumberF(v)      => G.point(NumberF(v))
+      case AddF(a, b)      => (f(a) ⊛ f(b))(AddF(_, _))
+      case SubtractF(a, b) => (f(a) ⊛ f(b))(SubtractF(_, _))
+      case MultiplyF(a, b) => (f(a) ⊛ f(b))(MultiplyF(_, _))
+      case DivideF(a, b)   => (f(a) ⊛ f(b))(DivideF(_, _))
+    }
+}
+```
+
+Now is when we start wanting to take advantage of Matryoshka, so let’s bring it into scope.
+
+```tut:book
+import matryoshka._
+import matryoshka.implicits._
+```
+
+Since you already have your recursive AST, you also have code that uses it, and the last thing you want to do when trying a new tool is have to rewrite the code you already have. So, we’ll keep your AST and your code – all you have to do is define the relationship between the original AST and the new one, which we’ll do like this:
+
+```tut:book
+implicit val arithmeticRecursive: Recursive.Aux[Arithmetic, ArithmeticF] =
+  Recursive.fromCoalgebra {
+    case Number(v)      => NumberF(v)
+    case Add(a, b)      => AddF(a, b)
+    case Subtract(a, b) => SubtractF(a, b)
+    case Multiply(a, b) => MultiplyF(a, b)
+    case Divide(a, b)   => DivideF(a, b)
+  }
+
+implicit val arithmeticCorecursive: Corecursive.Aux[Arithmetic, ArithmeticF] =
+  Corecursive.fromAlgebra {
+    case NumberF(v)      => Number(v)
+    case AddF(a, b)      => Add(a, b)
+    case SubtractF(a, b) => Subtract(a, b)
+    case MultiplyF(a, b) => Multiply(a, b)
+    case DivideF(a, b)   => Divide(a, b)
+  }
+```
+
+All that this does is define the mapping between the directly-recursive structure and the functor, so it’s just a matter of seeing where the `F` suffix is.
+
+**NB**: We _should_ be able to define this with a single `Birecursive.fromIsomorphism`, but `Birecursive` doesn’t currently work correctly (see #44).
+
+With this relationship established, we now have access to all the power of recursion schemes.
+
+```tut:book
+val pprint: Algebra[ArithmeticF, String] = {
+  case NumberF(v)      => v.toString
+  case AddF(a, b)      => "($a) + ($b)"
+  case SubtractF(a, b) => "($a) - ($b)"
+  case MultiplyF(a, b) => "($a) * ($b)"
+  case DivideF(a, b)   => "($a) / ($b)"
+}
+```
+
+```tut
+val expr: Arithmetic =
+  Add(Multiply(Number(3), Divide(Number(4), Number(5))), Number(6))
+expr.cata(pprint)
```

Comment:
When I `sbt tut`, it complains here that “value cata is not a member of Arithmetic”, but I _think_ I have all my imports right, and the implicits are at the top level now. Not sure what I’m missing.